contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3560,6 +3615,22 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 						else
 							orig_schema = "dbo";
 					}
+					else
+					{
+						owner_oid = rolspec ? get_rolespec_oid(rolspec, true) : InvalidOid;
```

Comment:
I think owner_oid should be `GetUserId()` when rolspec is invalid so that we create schema with current_user as owner.

Reply:
We are trying to check if current user is trying to create schema for some one else.
If the current user itself is the owner, rolspec will be null and `CreateSchemaCommand` will handle it.